### PR TITLE
Define index and create actions to rent tested 

### PR DESCRIPTION
**Pull request by lucasVoboril**
## Summary

The model **RENT** should have the followings routes: **index** and **create**. To do this it's necessary users had been authenticated. I've done tests for their respective routes

## Screenshots

It's not necessary
 
## Trello Card

https://trello.com/c/WRWzhCve

### CODE REVIEW

File: app/controllers/rents_controller.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+class RentsController < ApplicationController
+  include Wor::Paginate
+
+  before_action :authenticate_user!, only: %i[create index]
+
+  def create
+    rent = current_user.rent.build(create_params)
```
Comment (Martum) [inline app/controllers/rents_controller.rb]: I'm pretty sure this should be `current_user.rents.build(create_params)` with `rent` in plural!

File: spec/controllers/rents_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      it 'returns an unauthorized status' do
+        get :index
+        expect(response).to have_http_status(:unauthorized)
+      end
+    end
+
+    context 'With an authenticated user' do
+      include_context 'Authenticated User'
+
+      it 'without params returns all rents' do
```
Comment (Martum) [inline spec/controllers/rents_controller_spec.rb]: All examples should start with a verb. If you need to use `with`, `when` or `without` as your first word, maybe you are missing a context.

File: spec/controllers/rents_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        get :index
+        expect(JSON.parse(response.body)['total_count']).to eq(rents.size)
+      end
+
+      it 'returns a 200 ok status' do
+        get :index, params: { rent: { user: rents.first.user } }
+        expect(response).to have_http_status(:ok)
+      end
+
+      it 'incorrect params returns empty response' do
```
Comment (Martum) [inline spec/controllers/rents_controller_spec.rb]: Same here! Change this on the examples below!

File: app/controllers/rents_controller.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+            else
+              Rent.all
+            end
+    render_paginated rents, each_serializer: RentSerializer
+  end
+
+  private
+
+  def create_params
+    params.require(:rent).permit(:from, :to, :user_id, :book_id)
```
Comment (Martum) [inline app/controllers/rents_controller.rb]: We are using the `user_id` provided by `current_user`, so we shouldn't permit it.

File: spec/models/rent_spec.rb [hunk 5]
```diff
@@ -11,4 +11,11 @@
   it { is_expected.to validate_presence_of :to }
 
   it { expect(rent.from).to be <= rent.to }
+
+  it '#validate_date_from_is_less_than_date_to' do
```
Comment (Martum) [inline spec/models/rent_spec.rb]: Check what I said before about starting all examples with a verb! If you want to specify what's the method you are testing, use a `describe`. For example
```ruby
describe '#my_method'
  it 'returns X' do

  end
end
```

**Review by Martum — CHANGES_REQUESTED** [review]

File: app/controllers/rents_controller.rb (hunk 1 above)
Comment (lucasVoboril) [reply app/controllers/rents_controller.rb]: :thinking: it doesn't work
NoMethodError: undefined method `rents' for #<User:0x00005639ccec9008> 
Did you mean?  rent rent=

Comment (Martum) [reply app/controllers/rents_controller.rb]: We have a bug here https://github.com/wolox-training/lv-rails/blob/4b744c088381c8e0c6504ee625dbc2640fd08d27/app/models/user.rb#L12
It should be `:rents` in plural, as it is a collection of rents, not one!

Comment (lucasVoboril) [reply app/controllers/rents_controller.rb]: :sweat_smile: i thought has_many could automatic pluralize

**Review by Martum — APPROVED** [review]

**Pull request merged**